core/ui/src/main/kotlin/com/merxury/blocker/core/ui/applist/AppList.kt

```diff
@@ -57,13 +68,30 @@ fun AppList(
     onEnableClick: (String) -> Unit = {},
     onDisableClick: (String) -> Unit = {},
     onServiceStateUpdate: (String, Int) -> Unit = { _, _ -> },
+    onRefresh: () -> Unit = {},
 ) {
     val listState = rememberLazyListState()
     val scrollbarState = listState.scrollbarState(
         itemsAvailable = appList.size,
     )
+    var refreshing by remember { mutableStateOf(false) }
+    val scope = rememberCoroutineScope()
+    val refreshingState = rememberPullRefreshState(
+        refreshing = refreshing,
+        onRefresh = {
+            scope.launch {
+                refreshing = true
+                onRefresh()
```

Comment:
It indicates the `onRefresh` is a `suspend` function, or otherwise refreshing will immediately be set to false.